### ISSUE TITLE
Repository topics must be lowercase

### ISSUE THREAD

**Issue by martinssipenko** [terraform-providers/terraform-provider-github#165]
Github returns 422 error when topic is not lowercase. Works fine if topics are lowercased.

### Terraform Version
```
Terraform v0.11.8
+ provider.github v1.2.1
```

### Affected Resource(s)
- github_repository

**Comment by majormoses** [comment]
In the UI when you create one it downcases them for you so we should do the same in the provider.

**Comment by martinssipenko** [comment]
I tried doing that in code but failed as lowercasing topics prior sending them to github made terraform think there are unapplied changes. Thats because topics were with uppercase letters in tf file, but came back as lowercase from github.

**Comment by majormoses** [comment]
> I could push my non working code as a PR and perhaps with somone elses help we can get this fixed.

Sounds good, you probably just want to make the comparison also downcased as well. For example:
```golang
import {
    "strings"
}


if strings.ToLower("STRING_IN_STATE") == strings.ToLower("PROPOSED_STRING") {
    // trigger change
} else {
    // no-op
}

```

**Comment by radeksimko** [comment]
Since the API rejects uppercase I think the best way forward here would be to add a validation function which enforces topics to be lowercase.

We can raise the error during the validation phase, which is much earlier and give users more helpful feedback before `plan`ing or `apply`ing changes.

**Comment by ryndaniels** [comment]
The fix for this was released in the 2.1.0 version of the provider.
